### PR TITLE
Upgrade to Rails 7.1.

### LINKED ISSUE #769: Upgrade to Rails 7.1
Following the [Rails upgrade guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-7-0-to-rails-7-1), bump purl from Rails 7.0.x to 7.1.x.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #769

### CODE REVIEW

**Comment by cbeer** [comment]
Did you run `rails app:update`? I'm surprised there aren't the environment + config updates that usually come with a Rails bump.

**Comment by justinlittman** [comment]
Duh. No.

**Comment by lwrubel** [comment]
Just checking: is this ready for review now?

File: config/environments/development.rb [hunk 1]
```diff
@@ -56,6 +56,9 @@
   # Highlight code that triggered database queries in logs.
   # config.active_record.verbose_query_logs = true
 
+  # Highlight code that enqueued background job in logs.
+  config.active_job.verbose_enqueue_logs = true
```
Comment (jcoyne) [inline config/environments/development.rb:60]: We don't use active_job, so this should be commented out.

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by justinlittman** [comment]
Yes

File: config/environments/development.rb (hunk 1 above)
Comment (justinlittman) [reply config/environments/development.rb]: Commented out.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**